### PR TITLE
A28: Implemented GrayBMP.DrawThickLine (...)

### PR DESCRIPTION
**Pull request by tmmrajesh**

### CODE REVIEW

File: GrayBMP.cs [hunk 1]
```diff
@@ -126,6 +126,41 @@ public void Dirty ()
...
+         mPF.AddLine (ax, ay, bx, by);
+      }
+      mPF.Fill (this, gray);
+
+      void Arc (Point2 pt, double start, double end) {
+         double step = (end - start) / 10, theta = start;
+         for (int i = 0; i <= 10; i++, theta += step)
+            pts.Add (RadialMove (pt, width, theta).Round ());
+      }
+   }
```
Comment (tarydon) [inline GrayBMP.cs]: I think you can reduce this to just 4 points (60 degree steps) to cover the 180 degrees.

File: GrayBMP.cs [hunk 2]
```diff
@@ -126,6 +126,41 @@ public void Dirty ()
       End ();
    }
 
+   public void DrawThickLine (int x1, int y1, int x2, int y2, int width, int gray) {
```
Comment (tarydon) [inline GrayBMP.cs:129]: Your routine will draw a line with width = **width * 2** actually. Either scale the parameter inside the routine (preferred), or change the parameter name to halfWidth and make it a float / double.

File: PolyFillWin.cs [hunk 3]
```diff
@@ -54,7 +54,7 @@ class PolyFillWin : Window {
 
       foreach (var line in mDwg.Lines) {
          var ((x0, y0), (x1, y1)) = (line.A.Round (), line.B.Round ());
-         mBmp.DrawLine (x0, y0, x1, y1, 0);
+         mBmp.DrawThickLine (x0, y0, x1, y1, mScale / 2, 0);
```
Comment (tarydon) [inline PolyFillWin.cs]: Handle case of mScale=1 separately (just call DrawLine).

**Review by tarydon — CHANGES_REQUESTED** [review]

File: GrayBMP.cs (hunk 2 above)
Comment (tmmrajesh) [reply GrayBMP.cs:129]: Yes. Passing **mScale / 2** hid this error earlier. Thanks.

File: GrayBMP.cs (hunk 1 above)
Comment (tmmrajesh) [reply GrayBMP.cs]: Fixed. Simplified the _Cap_ drawing a little. With fixed step size, end angle was no longer needed.

**Pull request closed without merging**